kie-server-parent/kie-server-api/src/test/java/org/kie/server/api/marshalling/XStreamMarshallerTest.java

```diff
@@ -249,4 +253,30 @@ public void testUnmarshallDateObject() {
         assertEquals(LocalTime.of(10, 10, 10), dateObject.getLocalTime());
         assertEquals(OffsetDateTime.of(LocalDateTime.of(2017, 1, 1, 10, 10, 10), ZoneOffset.ofHours(1)), dateObject.getOffsetDateTime());
     }
+
+    @Test
+    public void testUnmarshallDateObjectWithExtraField() {
+        String expectedString = "<date-object>\n" +
+                                "  <localDate>2017-01-01</localDate>\n" +
+                                "  <localDateTime>2017-01-01T10:10:10</localDateTime>\n" +
+                                "  <localTime>10:10:10</localTime>\n" +
+                                "  <offsetDateTime>2017-01-01T10:10:10+01:00</offsetDateTime>\n" +
+                                "  <unknownField>jojojojo</unknownField>\n" +
+                                "</date-object>";
+        Set<Class<?>> extraClasses = new HashSet<>();
+        extraClasses.add(DateObject.class);
+        Marshaller marshaller = MarshallerFactory.getMarshaller(extraClasses, MarshallingFormat.XSTREAM, getClass()
+                .getClassLoader());
+        
+        assertThrows(UnknownFieldException.class, () -> marshaller.unmarshall(expectedString, DateObject.class));
+        DateObject dateObject = marshaller.unmarshall(expectedString, DateObject.class, Collections.singletonMap(
+                XStreamMarshaller.XSTREAM_IGNORE_UNKNOWN, Boolean.TRUE));
+        assertNotNull(dateObject);
+        assertEquals(LocalDate.of(2017, 1, 1), dateObject.getLocalDate());
+        assertEquals(LocalDateTime.of(2017, 1, 1, 10, 10, 10), dateObject.getLocalDateTime());
+        assertEquals(LocalTime.of(10, 10, 10), dateObject.getLocalTime());
+        assertEquals(OffsetDateTime.of(LocalDateTime.of(2017, 1, 1, 10, 10, 10), ZoneOffset.ofHours(1)), dateObject
+                .getOffsetDateTime());
+        assertThrows(UnknownFieldException.class, () -> marshaller.unmarshall(expectedString, DateObject.class));
```

Comment:
this was checked some lines above. Perhaps, were you trying to add the false flag?
```suggestion
        assertThrows(UnknownFieldException.class, () -> marshaller.unmarshall(expectedString, DateObject.class), Collections.singletonMap(XStreamMarshaller.XSTREAM_IGNORE_UNKNOWN, Boolean.FALSE));
```

Reply:
The repeteated class is that the flag is properly reset to false after a call with the flag equal to true, hence no need to test the false value